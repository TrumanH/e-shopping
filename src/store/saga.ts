import { categoriesSaga } from './categories/categories.saga';
import createSagaMiddleware from 'redux-saga';
import { all, call } from 'redux-saga/effects';

export function* rootSaga() {
    yield all([call(categoriesSaga)]);
  }

// create the saga middleware
export const sagaMiddleware = createSagaMiddleware();