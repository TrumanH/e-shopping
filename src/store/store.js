import { configureStore } from '@reduxjs/toolkit';
import userReducer from './user/user.slice';
import categoriesReducer from './categories/categories.slice';
import cartReducer from './cart/cart.slice';
// root reducer

export const store = configureStore({
  reducer: {
    user: userReducer,
    categories: categoriesReducer,
    cart: cartReducer,
  },
});