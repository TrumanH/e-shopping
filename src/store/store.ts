import userReducer from './user/user.slice';
import categoriesReducer from './categories/categories.slice';
import cartReducer from './cart/cart.slice';
import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { sagaMiddleware, rootSaga } from './saga';

const rootReducer = combineReducers({
  user: userReducer,
  categories: categoriesReducer,
  cart: cartReducer,
});

const persistConfig = {
  key: 'root',
  version: 1,
  storage,
  // blacklist: ['user'],
  whitelist: ['cart'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        // Ignore these action types
        ignoredActions: ['user/setUser', FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        // Ignore these field paths in all actions
        ignoredActionPaths: ['meta.arg', 'payload.timestamp'],
        // Ignore these paths in the state
        ignoredPaths: ['user.user'],
      },
    }).concat(sagaMiddleware),
  devTools: process.env.NODE_ENV !== 'production', // The Redux DevTools Extension is disabled for production
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

// then run the saga
sagaMiddleware.run(rootSaga)

export const persistor = persistStore(store);

// https://redux-toolkit.js.org/usage/usage-guide#use-with-redux-persist
