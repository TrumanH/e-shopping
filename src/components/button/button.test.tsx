import Button, { BUTTON_TYPES }  from './button.component';
import {render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {store} from "../../store/store";
import userEvent from "@testing-library/user-event";

describe('Button component', () => {

  const produceComponent = () => {
    return render(
      <Provider store={store}>
        <Button buttonType={BUTTON_TYPES.inverted}>Test</Button>
      </Provider>
    );
  }

  test('match the snapshot', ()=>{
    const { asFragment } = produceComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  test('render the button component', ()=>{
    produceComponent();
    expect(screen.getByText(/Test/i)).toBeInTheDocument();
  });

  test('click the button', ()=>{
    produceComponent();
    userEvent.click(screen.getByText(/Test/i));
  });
});

