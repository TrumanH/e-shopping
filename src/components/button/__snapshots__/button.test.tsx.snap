// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-dkrFOg sc-eDvSVe bAenDU iJPJvM"
  >
    Test
  </button>
</DocumentFragment>
`;
